Build: Bundled Minimap2 fails to compile on ARM64 macOS (Apple Silicon)

Hello QUAST team,

When running QUAST 5.3.0 on an ARM64 macOS system (Apple M1 Max, macOS 15.4.1), the bundled Minimap2 fails to compile during the "Running Contig analyzer" step. This prevents QUAST from using Minimap2 and limits its functionality.

The `quast.log` indicates:
--- QUAST.LOG SNIPPET ---
Running Contig analyzer...
Compiling Minimap2 (details are in quast/quast_libs/minimap2/make.log and make.err)
make -C <PATH_TO_QUAST_INSTALL_DIR>/quast_libs/minimap2 >> quast/quast_libs/minimap2/make.log 2>> quast/quast_libs/minimap2/make.err
The tool returned non-zero. See quast/quast_libs/minimap2/make.err for stderr.
WARNING: Failed to compile Minimap2 (quast/quast_libs/minimap2)! Try to compile it manually. ...
ERROR! Compilation of contig aligner software was unsuccessful! QUAST functionality will be limited.
--- END QUAST.LOG SNIPPET ---

The corresponding `make.err` in `quast/quast_libs/minimap2/` shows errors related to SSE intrinsics being incompatible with the ARM64 architecture:
--- MAKE.ERR SNIPPET ---
clang: warning: argument unused during compilation: '-msse2' [-Wunused-command-line-argument]
In file included from ksw2_ll_sse.c:9:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/emmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
    14 | #error "This header is only meant to be used on x86 and x64 architecture"
       |  ^
... 
make: *** [ksw2_ll_sse.o] Error 1
--- END MAKE.ERR SNIPPET ---

This issue appears to be due to SSE intrinsics in `ksw2_ll_sse.c` being incompatible with the ARM64 architecture.

It can be resolved by modifying QUAST's build process to pass ARM-specific compilation flags (`arm_neon=1 aarch64=1`) to Minimap2's standard Makefile when an ARM64 system is detected. ([See here](https://github.com/lh3/minimap2))
I have developed and tested a fix for this and would like to submit a Pull Request.

--- a/quast_libs/qutils.py
+++ b/quast_libs/qutils.py
@@ -13,6 +13,7 @@ import hashlib
 import shutil
 import subprocess
 import os
+import platform
 import stat
 import sys
 import re
@@ -892,10 +893,18 @@ def compile_tool(name, dirpath, requirements, just_notice=False, logger=logger, 
             call_subprocess(['./configure'] + configure_args, stdout=open(make_logs_basepath + '.log', 'a'),
                             stderr=open(make_logs_basepath + '.err', 'a'))
             os.chdir(prev_dir)
+
+        current_make_command_parts = (['make', make_cmd] if make_cmd else ['make'])
+        if name == 'Minimap2' and (platform.machine() == 'arm64' or platform.machine() == 'aarch64'):
+            logger.main_info(f'Detected 64-bit ARM ({platform.machine()}). Applying ARM-specific make flags for Minimap2.')
+            current_make_command_parts.extend(['arm_neon=1', 'aarch64=1'])
+
+        final_make_command_for_subprocess = current_make_command_parts + ['-C', dirpath]
+        
         try:
-            return_code = call_subprocess((['make', make_cmd] if make_cmd else ['make']) + ['-C', dirpath],
-                                      stdout=open(make_logs_basepath + '.log', 'a'),
-                                      stderr=open(make_logs_basepath + '.err', 'a'), logger=logger)
+            return_code = call_subprocess(final_make_command_for_subprocess,
+                                          stdout=open(make_logs_basepath + '.log', 'a'),
+                                          stderr=open(make_logs_basepath + '.err', 'a'), logger=logger)
         except IOError:
             msg = 'Permission denied accessing ' + dirpath + '. Did you forget sudo?'
             if just_notice:

